### ISSUE TITLE
version 2.5: unexpected EOF during plan in China region

### ISSUE THREAD

**Issue by kdubmartins** [terraform-providers/terraform-provider-aws#8265]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.10.8

### Affected Resource(s)

* aws_spot_fleet_request
* aws_security_group
* terraform plan

### Terraform Configuration Files

```hcl
provider "aws" {
  region              = "cn-north-1"
  profile             = "my-china-profile"
  allowed_account_ids = ["00000000"]

  #https://github.com/terraform-providers/terraform-provider-aws/issues/4690
  #https://github.com/terraform-providers/terraform-provider-aws/issues/4967

  # this is not picked up
  endpoints = {
    autoscaling = "https://autoscaling.cn-north-1.amazonaws.com.cn"
  }

  # so we need this for the hack in the etc hosts
  insecure = true
}
```

### Debug Output

https://gist.github.com/kdubmartins/772ec7ac1b87c74bebb645662371b51f

### Expected Behavior

A successful `terraform plan` output.

### Actual Behavior

```
Error: Error refreshing state: 2 error(s) occurred:

* module.nomad_sg.provider.aws: unexpected EOF
* module.nomad.provider.aws: unexpected EOF
```

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

Adding a constraint to `version = "~> 2.4"` fixes the issue and the plan runs just fine.
Also, it does not happen in other AWS regions like eu-west-1 or us-east-1.

**Comment by nywilken** [comment]
@kdubmartins thank you for reporting this issue. This bug was introduced in version 2.5.0 of the AWS provider. A fix has been submitted in #8226, which will be in the next release. 

For now, as you specified, pinning the provider version to 2.4.0 should help get around the issue.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.6.0 of the Terraform AWS Provider later today. 👍

**Comment by bflad** [comment]
The fix for this has been released in [version 2.6.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#260-april-10-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
